### ISSUE TITLE
Open editable files in read-only mode

### ISSUE THREAD

**Issue by pmarini-nc** [nextcloud/server#29783]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**

When opening editable files in the browser for which the user has write privileges, it would be useful to give the possibility to have it opened in read-only mode. In many use cases, this is what users most likely want to do so it could be even considered as the default opening mode.

The files that I'm thinking of are:

- text and markdown files that can be edited in the browser with Nextcloud Text
- spreadsheets, presentations and word documents that can be edited in the browser if an Office Suite service is configured

When a writable file is opened, the general behavior is the cursor to be placed in the document so it is sufficient that the user press Enter and the document is modified inadvertently. 

It is true that older versions can be recovered but it can be a complex operation if several users access the same file around the same time and make an unwanted modification. Moreover we cannot currently label/comment version so that the user couldn't write something like "Restoring previous version for unintended modification".

**Describe the solution you'd like**
It could be a button for example in the side bar or the default opening mode. 

**Describe alternatives you've considered**
There are no alternatives as per my knowledge, just be extra-careful when opening a writable file that I just want to read. 

**Additional context**

**Comment by szaimen** [comment]
cc @nextcloud/office

**Comment by pmarini-nc** [comment]
Before closing this ticket, I would like to have the feedback from @juliushaertl

**Comment by juliushaertl** [comment]
I'm not sure how common that use case is and if that would be something we should generally introduce for file actions. @jancborchardt @nimishavijay Do you think this is something we should have?

**Comment by jancborchardt** [comment]
Yes, also not sure here. I only remember it from Google Docs on mobile, where you get a floating edit button on the bottom right – and we do it the same in our mobile apps.

But if you are on a computer, we shouldn't have modes here as people will be confused about why they can't edit, and/or ask their colleagues to check the permissions.

> In many use cases, this is what users most likely want to do so it could be even considered as the default opening mode.

What are some of these use cases?

**Comment by pmarini-nc** [comment]
Thanks @juliushaertl and @jancborchardt for your valuable feedback.

I was thinking to the fact that many times you just want to read a document, not modify it. If a document is in a group folder or shared with many other users an unwanted modification would bring confusion to the other users.

Here are some considerations / remarks:

- desktop document editors, being them for simple or rich text, always ask for the user confirmation before saving on disk. This is not the case with Nextcloud Text, not sure about other tools, where autosave is in place.
- for ignorance or just convenience, editable files are often shared in read/write mode even if the owner doesn't really want them to be editable
- in Google Docs there are three document roles: editor, commenter and viewer with clear rights on the document. This could also be something helpful when collaborating.

**Comment by pmarini-nc** [comment]
This issue shouldn't be closed as I'm still waiting for a feedback from @juliushaertl and @jancborchardt

**Comment by jancborchardt** [comment]
> desktop document editors, being them for simple or rich text, always ask for the user confirmation before saving on disk. This is not the case with Nextcloud Text, not sure about other tools, where autosave is in place.

Google Docs for example autosaves though. And iOS Notes autosaves as well. This is quite a common pattern and only on the rise.

> in Google Docs there are three document roles: editor, commenter and viewer with clear rights on the document. This could also be something helpful when collaborating.

This already exists in Nextcloud via the share permissions, which control whether someone can edit or not.

> for ignorance or just convenience, editable files are often shared in read/write mode even if the owner doesn't really want them to be editable

This is something which could be fixed, and this is tackled by the clearer "Can edit" and "View only" sharing indicator in the subline of the new design for sharing: https://github.com/nextcloud/server/issues/26691

**Comment by pmarini-nc** [comment]
What I'm referring to here, is not for the person that shares the file but for the recipient. 

Imagine the following use case:

Alice has worked on a new document and has a first draft ready to be shared with the team. Alice puts the document in the appropriate Group Folder and communicate in some way to Bob and Carl, that are in the same group, about it. The Group Folder is the appropriate place to store the document as she doesn't want to keep a separate copy and all the group members have write privileges in it - as it should be. Bob and Carl open the document and by error change the document. This is very easy as, as I mentioned above, the cursor is placed in the document and just pressing ENTER triggers a modification. Now there is a potentially messy situation as Bob and Carl try to recover a previous version, that is actually not that easy - as versions are not labeled.

This scenario could be avoided by forcing the View Mode on document opening. If the user wants to modify the document and he/she is entitled to do so, he/she can activate the Edit mode.

I understand that there may be also some drawbacks with this choice, but I think my point should now be clear.

**Comment by azul** [comment]
Nextcloud Collectives displays documents in read only mode and then has an edit button. The idea here being that a collective is a collection of texts and the most common usecase will actually be to read them - like in a handbook or so.
The next version of collectives will include the ability to share collectives with links. So if you are looking for a way to share your texts editable but with an explicit edit button this might be an option.

**Comment by pmarini-nc** [comment]
Ok thanks I'll have a look!

**Comment by pmarini-nc** [comment]
ping

**Comment by juliushaertl** [comment]
I think opening files in edit mode is rather the expected behaviour around different solutions, I'd say safeguarding unintentional user changes is something where existing rollback solutions like undo and versioning should be sufficient already. And it would safe us additional complexity in the file opening behaviour and another case in the qa matrix.

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.
